src/caps.rs

```diff
@@ -71,87 +104,38 @@ pub struct CodeAssistantCaps {
-        caps_urls.push("<compiled-in-caps-hf>".to_string());
-    } else {
-        if cmdline.address_url.starts_with("http") {
-            is_remote_address = true;
-            let base_url = Url::parse(&cmdline.address_url.clone()).map_err(|_| "failed to parse address url (1)".to_string())?;
-            let joined_url = base_url.join(&CAPS_FILENAME).map_err(|_| "failed to parse address url (2)".to_string())?;
-            let joined_url_fallback = base_url.join(&CAPS_FILENAME_FALLBACK).map_err(|_| "failed to parse address url (2)".to_string())?;
-            caps_urls.push(joined_url.to_string());
-            caps_urls.push(joined_url_fallback.to_string());
-        } else {
-            is_local_file = true;
-            caps_urls.push(cmdline.address_url.clone());
-        }
-    }
-    let caps_url: String = match caps_urls.get(0) {
-        Some(u) => u.clone(),
-        None => return Err("caps_url is none".to_string())
-    };
-    if is_local_file {
-        let mut file = File::open(caps_url.clone()).map_err(|_| format!("failed to open file '{}'", caps_url))?;
-        file.read_to_string(&mut buffer).map_err(|_| format!("failed to read file '{}'", caps_url))?;
-    }
-
-    let http_client = global_context.read().await.http_client.clone();
-    let api_key = cmdline.api_key.clone();
-    let mut headers = reqwest::header::HeaderMap::new();
-    if !api_key.is_empty() {
-        headers.insert(reqwest::header::AUTHORIZATION, reqwest::header::HeaderValue::from_str(format!("Bearer {}", api_key).as_str()).unwrap());
-        headers.insert(reqwest::header::USER_AGENT, reqwest::header::HeaderValue::from_str(format!("refact-lsp {}", crate::version::build_info::PKG_VERSION).as_str()).unwrap());
-    }
     let mut r1_mb_error_text = "".to_string();
-
-    if is_remote_address {
-        let mut status: u16 = 0;
-        for url in caps_urls.iter() {
-            info!("fetching caps from {}", url);
-            let response = http_client.get(url).headers(headers.clone()).send().await.map_err(|e| format!("{}", e))?;
-            status = response.status().as_u16();
-            buffer = match response.text().await {
-                Ok(v) => v,
-                Err(_) => continue
-            };
-
-            if status == 200 {
-                break;
-            }
-
-            warn!("status={}; server responded with:\n{}", status, buffer);
-        }
-        if status != 200 {
-            return Err(format!("cannot fetch caps, status={}", status));
-        }
-    }
-
+    
     let r1_mb: Option<CodeAssistantCaps> = match serde_json::from_str(&buffer) {
         Ok(v) => v,
         Err(e) => {
-            r1_mb_error_text = format!("{}", e);
-            None
+            match serde_yaml::from_str(&buffer) { 
+                Ok(v) => v,
+                Err(e) => {
+                    r1_mb_error_text = format!("{}", e);
+                    None
+                }
+            }
```

Comment:
Check for "{" as a first character instead. This code doesn't handle json with an error well.